tiny_cnn/core/nnp_backend.h

```diff
@@ -56,40 +58,40 @@ class nnp_backend : public math_backend {
       const nnp_convolution_kernel_transform_strategy kernel_tf_strategy =
           nnp_convolution_kernel_transform_strategy_reuse;
 
-      const cnn_size_t input_channels = this->in_.depth_;
-      const cnn_size_t output_channels = this->out_.depth_;
+      const cnn_size_t input_channels = params_->in.depth_;
+      const cnn_size_t output_channels = params_->out.depth_;
 
       const nnp_size input_size = {
-          static_cast<size_t>(this->in_.width_),
-          static_cast<size_t>(this->in_.height_)
+          static_cast<size_t>(params_->in.width_),
+          static_cast<size_t>(params_->in.height_)
       };
 
       const nnp_size kernel_size = {
-          static_cast<size_t>(this->weight_.width_),
-          static_cast<size_t>(this->weight_.height_)
+          static_cast<size_t>(params_->weight.width_),
+          static_cast<size_t>(params_->weight.height_)
       };
 
-      /*const nnp_padding padding = {.top = static_cast<size_t>(pad.cpu_data()[0]),
-                                   .right = static_cast<size_t>(pad.cpu_data()[1]),
-                                   .bottom = static_cast<size_t>(pad.cpu_data()[0]),
-                                   .left = static_cast<size_t>(pad.cpu_data()[1])};*/
-      const nnp_padding padding = { 0, 0, 0, 0 };
+      const nnp_padding padding = {
+          static_cast<size_t>(params_->in_padded.width_),
+          static_cast<size_t>(params_->in_padded.height_),
+          static_cast<size_t>(params_->in_padded.height_),
+          static_cast<size_t>(params_->in_padded.width_)
+      };
 
       const float* input_pointer =
-          reinterpret_cast<const float*>(
-              this->conv_layer_worker_storage_[index].prev_out_padded_);
+          reinterpret_cast<const float*>(&in_data[0]->at(0));
 
       const float* kernel_pointer =
-          reinterpret_cast<const float*>(&in_data[0]->at(0));
+          reinterpret_cast<const float*>(&in_data[1]->at(0));
 
       const float* bias =
-          reinterpret_cast<const float*>(&in_data[1]->at(0));
+          reinterpret_cast<const float*>(&in_data[2]->at(0));
 
```

Comment:
@edgarriba 

- Let's change the following line in config.h to correct type mismatch.
```cpp
//typedef double float_t;
typedef float float_t
```

- ```in_padded``` in tiny-cnn is size of input images after padding, and ```nnp_padding``` in NNPack is size of padding itself (in_padded.width = in.width + padding.left + padding.right). so value of ```padding``` should be corrected.

- Let's change the index of ```output_pointer```. out_data[0] is used for the output data after activation. And I think moving these code to convolutional_layer is better. (These indexes depend on the implementation detail of convolutional layer, so the implementation of the layer and the engine is tightly coupled in current design)

```cpp
      //float* output_pointer = reinterpret_cast<float*>(&out_data[0]->at(0));
      float* output_pointer = reinterpret_cast<float*>(&out_data[1]->at(0));
```